rxandroid/src/test/java/io/reactivex/android/schedulers/HandlerSchedulerTest.java

```diff
@@ -0,0 +1,807 @@
+        assertSame(counter, runnableRef.get());
+
+        runUiThreadTasks();
+        // Verify the scheduled runnable was the one returned from the hook.
+        assertEquals(1, newCounter.get());
+        assertEquals(0, counter.get());
+    }
+
+    @Test
+    public void directScheduleDisposedDoesNotRun() {
+        CountingRunnable counter = new CountingRunnable();
+        Disposable disposable = scheduler.scheduleDirect(counter);
+        disposable.dispose();
+
+        runUiThreadTasks();
+        assertEquals(0, counter.get());
+    }
+
+    @Test
+    public void directScheduleWithDelayPostsWithDelay() {
+        CountingRunnable counter = new CountingRunnable();
+        scheduler.scheduleDirect(counter, 1, MINUTES);
+
+        runUiThreadTasks();
+        assertEquals(0, counter.get());
+
+        idleMainLooper(1, MINUTES);
+        runUiThreadTasks();
+        assertEquals(1, counter.get());
+    }
+
+    @Test
+    public void directScheduleWithDelayUsesHook() {
+        final CountingRunnable newCounter = new CountingRunnable();
+        final AtomicReference<Runnable> runnableRef = new AtomicReference<>();
+        RxJavaPlugins.setScheduleHandler(new Function<Runnable, Runnable>() {
+            @Override public Runnable apply(Runnable runnable) {
+                runnableRef.set(runnable);
+                return newCounter;
+            }
+        });
+
+        CountingRunnable counter = new CountingRunnable();
+        scheduler.scheduleDirect(counter, 1, MINUTES);
+
+        // Verify our runnable was passed to the schedulers hook.
+        assertSame(counter, runnableRef.get());
+
+        idleMainLooper(1, MINUTES);
+        runUiThreadTasks();
+        // Verify the scheduled runnable was the one returned from the hook.
+        assertEquals(1, newCounter.get());
+        assertEquals(0, counter.get());
+    }
+
+    @Test
+    public void directScheduleWithDelayDisposedDoesNotRun() {
+        CountingRunnable counter = new CountingRunnable();
+        Disposable disposable = scheduler.scheduleDirect(counter, 1, MINUTES);
+
+        idleMainLooper(30, SECONDS);
+        disposable.dispose();
+
+        idleMainLooper(30, SECONDS);
+        runUiThreadTasks();
+        assertEquals(0, counter.get());
+    }
+
+    @Test
+    public void directSchedulePeriodicallyReschedulesItself() {
+        CountingRunnable counter = new CountingRunnable();
+        scheduler.schedulePeriodicallyDirect(counter, 1, 1, MINUTES);
+
+        runUiThreadTasks();
+        assertEquals(0, counter.get());
+
+        idleMainLooper(1, MINUTES);
+        runUiThreadTasks();
+        assertEquals(1, counter.get());
+
+        idleMainLooper(1, MINUTES);
+        runUiThreadTasks();
+        assertEquals(2, counter.get());
+
+        idleMainLooper(1, MINUTES);
+        runUiThreadTasks();
+        assertEquals(3, counter.get());
+    }
+
+    @Test
+    public void directSchedulePeriodicallyUsesHookOnce() {
```

Comment:
Actually, Scheduler's method is `schedulePeriodicallyDirect()`, probably worth to update tests names. 

Same is applicable for `scheduleDirect()`